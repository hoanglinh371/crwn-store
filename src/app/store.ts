import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import cartReducer from '../features/cart/redux/cart.slice';
import categoryReducer from '../features/category/redux/category.slice';
import productReducer from '../features/product/redux/product.slice';

export const store = configureStore({
  reducer: {
    productReducer,
    cartReducer,
    categoryReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
