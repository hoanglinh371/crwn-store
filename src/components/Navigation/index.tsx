import React from 'react';
import classNames from 'classnames/bind';
import { Link } from 'react-router-dom';
import { ReactComponent as Logo } from 'src/assets/crown.svg';

import styles from './Navigation.module.scss';

import CartIcon from '../CartIcon';
import CartDropdown from '../CartDropdown';

const cx = classNames.bind(styles);

const Navigation: React.FC = () => {
  return (
    <div className={cx('navigation')}>
      <Link className={cx('logo-container')} to='/'>
        <Logo />
      </Link>
      <div className={cx('nav-links-container')}>
        <Link className={cx('nav-link')} to='/shop'>
          SHOP
        </Link>
        <Link className={cx('nav-link')} to='/contact'>
          CONTACT
        </Link>
        <Link className={cx('nav-link')} to='/auth'>
          LOGIN
        </Link>
        <CartIcon />
      </div>
      <CartDropdown />
    </div>
  );
};

export default Navigation;
