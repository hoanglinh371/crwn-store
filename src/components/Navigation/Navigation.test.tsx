import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from 'src/app/store';
import Navigation from '.';

describe('Navigation component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
