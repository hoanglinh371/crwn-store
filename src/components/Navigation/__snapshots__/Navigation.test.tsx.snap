// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="navigation"
  >
    <a
      class="logo-container"
      href="/"
    >
      <svg>
        crown.svg
      </svg>
    </a>
    <div
      class="nav-links-container"
    >
      <a
        class="nav-link"
        href="/shop"
      >
        SHOP
      </a>
      <a
        class="nav-link"
        href="/contact"
      >
        CONTACT
      </a>
      <a
        class="nav-link"
        href="/auth"
      >
        LOGIN
      </a>
      <div
        class="container"
      >
        <svg
          class="shopping-icon"
        >
          shopping-bag.svg
        </svg>
        <span
          class="item-count"
        >
          0
        </span>
      </div>
    </div>
  </div>
</div>
`;
